### PR TITLE
Update default template

### PR DESCRIPTION
**Pull request by ratik21**
Updated `/default` template ASA transfer functionality. Attached screenshots.

image
image
image
image
image
image
image
image
image

NOTE: i don't think there is an Algosigner API method for selecting default account. If there are multiple accounts added by user in his wallet, then user will need to select from account address, from one the `algosigner.accounts` (choose from wallet.accounts in a dropdown).

### CODE REVIEW

File: default/scripts/1-tesla-asa.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  john is transferred some funds to execute opt-in transaction
+*/
+async function run(runtimeEnv, deployer) {
+  console.log('[tesla]: Script has started execution!');
+
+  const masterAccount = deployer.accountsByName.get('master-account');
+  const elon = deployer.accountsByName.get('elon-musk');
+  const john = deployer.accountsByName.get('john');
+
+  // activate elon account
```
Comment (amityadav0) [inline default/scripts/1-tesla-asa.js]:
```suggestion
```

File: default/scripts/1-tesla-asa.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  );
+
+  const asaInfo = await deployer.deployASA('tesla', { creator: elon });
+  console.log(asaInfo);
+
+  await deployer.optInAcountToASA(asaInfo.assetIndex, 'john', {});
+
+  const assetID = asaInfo.assetIndex;
+  await balanceOf(deployer, elon.addr, assetID);
+  // await balanceOf(deployer, john.addr, assetID);
```
Comment (amityadav0) [inline default/scripts/1-tesla-asa.js]:
```suggestion
```

File: default/src/components/signer.js [hunk 3]
```diff
@@ -126,126 +147,279 @@ const GetTxParams = () => {
...
   let deployedASANames = [];
-  let deployedASAIds = [];
-  let asaId;
-  for (const [key, value] of Object.entries(asaData)) {
-    deployedASANames.push(key);
-    deployedASAIds.push(value.assetIndex);
+  let asaInfo = {};
+  const assets = Object.entries({ ...goldAsaInfo, ...teslaAsaInfo });
+  for (const [name, info] of assets) {
+    asaInfo[name] = info;
```
Comment (amityadav0) [inline default/src/components/signer.js:158]: can we use `assets` above instead of asaInfo ?

File: default/scripts/1-tesla-asa.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  console.log('[tesla]: Script execution has finished!');
+}
+
+module.exports = { default: run };
```
Comment (amityadav0) [inline default/scripts/1-tesla-asa.js:37]: could you please help me understand how are we using this file in d-app ?

File: default/src/components/signer.js (hunk 3 above)
Comment (ratik21) [reply default/src/components/signer.js:158]: No, assets is not a map, so had to construct one.
image

File: default/scripts/1-tesla-asa.js (hunk 4 above)
Comment (ratik21) [reply default/scripts/1-tesla-asa.js:37]: Yes, in dapp we're using deployed asset info (from `/artifacts`). So in the UI dropdown we have an option to choose "tesla" asset (deployed via this script), and transfer it using algosigner wallet

**Review by robert-zaremba — APPROVED** [review]
looks much better

**Pull request merged**